ssc/cmod_windpower.cpp

```diff
@@ -377,26 +408,35 @@ void cm_windpower::exec()
 		assign("annual_gross_energy", gross_energy);
 		assign("wind_speed", avg_speed);
         calculate_p50p90(this);
-        calculate_losses(this, wake_int_loss_percent);
+        calculate_losses(this, annual_wake_int_loss_percent);
 		return;
 	}
 
     // create wakeModel
     std::shared_ptr<wakeModelBase> wakeModel(nullptr);
     int wakeModelChoice = as_integer("wind_farm_wake_model");
-    if (wakeModelChoice == 0)
+    if (wakeModelChoice == SIMPLE)
         wakeModel = std::make_shared<simpleWakeModel>(simpleWakeModel(wpc.nTurbines, &wt));
-    else if (wakeModelChoice == 1)
-        wakeModel = std::make_shared<parkWakeModel>(parkWakeModel(wpc.nTurbines, &wt));
-    else if (wakeModelChoice == 2)
+    else if (wakeModelChoice == PARK)
+    {
+        // get optional wake decay constant
+        double wdc = 0.07; //this is the default value
+        if (is_assigned("park_wake_decay_constant"))
+            wdc = as_double("park_wake_decay_constant");
+
+        // create the wake model
+        wakeModel = std::make_shared<parkWakeModel>(parkWakeModel(wpc.nTurbines, &wt, wdc));
+    }
+    else if (wakeModelChoice == EDDYVISCOSITY)
     {
         wpc.turbulenceIntensity *= 100;
         wakeModel = std::make_shared<eddyViscosityWakeModel>(eddyViscosityWakeModel(wpc.nTurbines, &wt, as_double("wind_resource_turbulence_coeff")));
     }
-    else if (wakeModelChoice == 3)
+    else if (wakeModelChoice == CONSTANTVALUE)
     {
-        wake_int_loss_percent = as_double("wake_int_loss");
-        wakeModel = std::make_shared<constantWakeModel>(constantWakeModel(wpc.nTurbines, &wt, (100. - wake_int_loss_percent)/100.));
+        annual_wake_int_loss_percent = as_double("wake_int_loss");
+        annual_wake_int_loss_percent *= is_assigned("wake_loss_multiplier") ? as_double("wake_loss_multiplier") : 1.; // add any wake loss multipliers to the constant loss value here
+        wakeModel = std::make_shared<constantWakeModel>(constantWakeModel(wpc.nTurbines, &wt, (100. - annual_wake_int_loss_percent)/100.));
```

Comment:
Am I understanding that for the constant wake loss model you can provide both the constant and some multiple of the constant? e.g. loss = a * b? 

I think it would make more sense to just provide one value, and make "wake_loss_multiplier" not applicable for wakeModelChoice == CONSTANTVALUE

Reply:
You're right, that would be pretty confusing. Updated!